include/boost/dll/shared_library.hpp

```diff
@@ -511,6 +511,26 @@ class shared_library
         return base_t::suffix();
     }
 
+    /*!
+    * Returns the decorated path to a shared module name, i.e. with needed prefix/suffix added.
```

Comment:
Need more scary notes:
`Returns the decorated path to a shared module name for the current runtime environment, i.e. ...`
`Calling this function with different current directories may produce different results.`

Reply:
Ther result of `shared_library::decorate("foo")` may not be the same, if user changes the current directory via https://en.cppreference.com/w/cpp/filesystem/current_path
For example it may return linfoo.so and foo.so, depending on the working directory.

Oh, one more thing: please add a test, that if we create an empty file with name `test_broken.so` and try to load it, then the the loading fails. You may also write some other tests.